spacy/pipeline/span_finder.py

```diff
@@ -0,0 +1,389 @@
+from functools import partial
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, cast
+
+from thinc.api import (Config, Model, Ops, Optimizer, get_current_ops,
+                       set_dropout_rate)
+from thinc.types import Floats2d, Ints1d, Ragged
+
+from spacy.language import Language
+from spacy.pipeline.trainable_pipe import TrainablePipe
+from spacy.scorer import Scorer
+from spacy.tokens import Doc
+from spacy.training import Example
+
+from ..util import registry
+from .spancat import Suggester
+
+span_finder_default_config = """
+[model]
+@architectures = "spacy.SpanFinder.v1"
+
+[model.scorer]
+@layers = "spacy.LinearLogistic.v1"
+nO = 2
+
+[model.tok2vec]
+@architectures = "spacy.Tok2Vec.v2"
+
+[model.tok2vec.embed]
+@architectures = "spacy.MultiHashEmbed.v2"
+width = 96
+rows = [5000, 2000, 1000, 1000]
+attrs = ["ORTH", "PREFIX", "SUFFIX", "SHAPE"]
+include_static_vectors = false
+
+[model.tok2vec.encode]
+@architectures = "spacy.MaxoutWindowEncoder.v2"
+width = ${model.tok2vec.embed.width}
+window_size = 1
+maxout_pieces = 3
+depth = 4
+"""
+
+DEFAULT_SPAN_FINDER_MODEL = Config().from_str(span_finder_default_config)["model"]
+DEFAULT_PREDICTED_KEY = "span_candidates"
+# XXX What was this TODO for?
+DEFAULT_TRAINING_KEY = "sc"  # TODO: define in spancat
```

Comment:
Use the value from `spancat` instead, or if it's not defined separately there, define it there and import it here.